job-dsl-core/src/main/groovy/javaposse/jobdsl/dsl/helpers/step/StepContext.groovy

```diff
@@ -42,6 +42,27 @@ class StepContext extends AbstractExtensibleContext {
     }
 
     /**
+     * Runs a xshell script.
+     *
+     * @since 1.41
+     */
+    @RequiresPlugin(id = 'xshell', minimumVersion = '0.10')
+    void xshell(@DslContext(XShellContext) Closure xShellClosure) {
+        XShellContext xShellContext = new XShellContext(jobManagement)
+        ContextHelper.executeInContext(xShellClosure, xShellContext)
+        stepNodes << new NodeBuilder().'hudson.plugins.xshell.XShellBuilder' {
+            delegate.executableInWorkspaceDir(xShellContext.executableInWorkspaceDir)
```

Comment:
The XML element is still called `executeFromWorkingDir`. Do not rename the XML element names. DSL methods names and XML element names must not match. Use sane name for the DSL methods using the UI labels as an orientation. The XML element names are just implementation details. And please test your changes before pushing.